src/content/docs/en/guides/testing.mdx

```diff
@@ -16,6 +16,132 @@ A Vite-native unit test framework with ESM, TypeScript and JSX support powered b
+  <Fragment slot="yarn">
+  ```shell
+  yarn add cypress --dev
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Configuration
+
+In the root of your project, create a `cypress.config.js` file with the following content:
+
+```js title="cypress.config.js"
+import { defineConfig } from 'cypress'
+
+export default defineConfig({
+  e2e: {
+    supportFile: false
+  }
+})
+```
+
+### Create your first Cypress test
+
+1. Choose a page to test. This example will test the example page `index.astro` below.
+
+    ```html title="src/pages/index.astro"
+    ---
+    ---
+    <html lang="en">
+      <head>
+        <title>Astro is awesome!</title>
+        <meta name="description" content="Pull content from anywhere and serve it fast with Astro's next-gen island architecture." />
+      </head>
+      <body>
+      <h1>Hello world from Astro</h1>
+      </body>
+    </html>
+    ```
+
+2. Create an `index.spec.js` file in the `cypress/e2e` folder. Use the following test in the file to verify that the page title and header are correct.
+
+    ```js title="cypress/e2e/index.cy.ts"
+    it('titles are correct', () => {
+      const page = cy.visit('http://localhost:3000');
+
+      page.get('title').should('have.text', 'Astro is awesome!')
+      page.get('h1').should('have.text', 'Hello world from Astro');
+    });
+    ```
+
+    :::tip[set a baseUrl]
+    You can set [`"baseUrl": "http://localhost:3000"`](https://docs.cypress.io/guides/end-to-end-testing/testing-your-app#Step-3-Configure-Cypress) in the `cypress.config.ts` configuration file to use `cy.visit("/")` instead of `page.goto("http://localhost:3000/")` for a more convenient URL.
+    :::
+
+### Running your Cypress tests
+
+Cypress can be run from the command line or from the Cypress App. The App provides a visual interface for running and debugging your tests.
+
+First, start the dev server so Cypress can access your live site.
+
+To run our test from the previous example using the command line, execute the following command:
+
+```shell
+npx cypress run
+```
+
+Alternatively, to run the test using the Cypress App, execute the following command:
+
+```shell
+npx cypress open
+```
+
+Once the Cypress App is launched, choose **E2E Testing**, then select the browser to be used to run tests.
+
+Once the test run is finished, you should see green check marks in the output confirming that your test passed:
+
+```shell title="Output from npx cypress run"
+Running:  index.cy.ts                                                                     (1 of 1)
+
+
+✓ titles are correct (107ms)
+
+1 passing (1s)
+```
+
+:::note[fail the test]
+To check that your test really does work, you can change the following line in the `index.astro` file:
+
+ ```astro title="src/pages/index.astro" del={2} ins={3}
+  <body>
+    <h1>Hello world from Astro</h1>
+    <h1>Hello from Astro</h1>
+  </body>
+```
+
+Then run the test again. You should see a red mark in the output confirming that your test failed.
```

Comment:
```suggestion
Then run the test again. You should see a red "X" in the output confirming that your test failed.
```
I'm not sure whether this is actually an X, but I think the very last thing here is to also be explicit about what kind of mark a failing test produces. It's helpful to describe the mark with more than just its colour, considering colourblindness etc.

Reply:
Yes, that's a good point, I haven't considered it yet.